.github/workflows/node-build.yml

```diff
@@ -0,0 +1,316 @@
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+      - name: Configure git
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+      - id: version-generator
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          VERSION_PREFIX=$(echo "${{ github.ref_name }}" | sed 's|release/||')
+          read major minor patch pre_release <<< $(echo "$VERSION_PREFIX" | awk -F'[.v-]' '{print $2, $3, $4, $5}')
+
+          version_search="v$major.$minor.*"
+
+          if [ -n "$pre_release" ]; then
+              version_search="$version_search-$pre_release"
+          fi
+          echo "VERSION_SEARCH: $version_search"
+          VERSION_PREFIX=$(git tag -l $version_search --sort=-taggerdate | head -n 1)
+
+          if [ -n "$VERSION_PREFIX" ]; then
+            read major minor patch pre_release <<< $(echo "$VERSION_PREFIX" | awk -F'[.v-]' '{print $2, $3, $4, $5}')
+            patch=$((patch + 1))
+          fi
+
+          NEW_VERSION="v${major}.${minor}.${patch}"
+
+          if [ -n "$pre_release" ]; then
+            NEW_VERSION="$NEW_VERSION-${pre_release}"
+          fi
+
+          echo "NEW_VERSION=$NEW_VERSION" >> $GITHUB_OUTPUT
+          echo "New version will be: $NEW_VERSION"
+
+          git tag -a $NEW_VERSION -m "$NEW_VERSION"
+          git push origin $NEW_VERSION
+  
+  docker-build:
+    runs-on: ubuntu-latest
+    needs: version-generator
+    timeout-minutes: 5
+    strategy:
+      matrix:
+        package: [auth, backend, frontend]
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Build
+        uses: docker/build-push-action@v5
+        with:
+          push: false
+          platforms: linux/amd64,linux/arm64
+          file: packages/${{ matrix.package }}/Dockerfile
+          tags: ghcr.io/${{ github.repository_owner }}/rafiki-${{ matrix.package }}:${{ needs.version-generator.outputs.version }}
+          outputs: type=docker,dest=/tmp/${{ matrix.package }}.tar
+      - name: Save docker image to cache
+        uses: actions/cache@v4
+        with:
+          path: /tmp/${{ matrix.package }}.tar
+          key: ${{ github.sha }}
+  
+  docker-grype:
+    name: Docker Grype Scan
+    needs: docker-build
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    strategy:
+      matrix:
+        package: [auth, backend, frontend]
+    steps:
+      - name: Fetch docker image from cache
+        uses: actions/cache/restore@v4
+        with:
+          path: /tmp/${{ matrix.package }}.tar
+          key: ${{ github.sha }}
+      - name: Scan docker image
+        uses: anchore/scan-action@v3
+        with:
+          image: /tmp/${{ matrix.package }}.tar
+          fail-build: true
+          only-fixed: true
+          severity-cutoff: high
+          output-format: table
+
+  docker-trivy:
```

Comment:
grype and trivy: how much do they overlap? ie is it good to keep both vs just picking one?

Reply:
Well, the difference is basically in the data sources that the scanner is using. Trivy is using more data sorces than grype, however, grype is using some of the data sources that trivy is not. So basically, we can keep both, or at least, just Trivy. Your thought?

Reply:
might as well have a broader check, we can keep both - just wanted to understand the differences